src/test/java/org/takes/facets/slf4j/BkLoggedTest.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.takes.facets.slf4j;
+
+import java.io.IOException;
+import java.net.Socket;
+import org.apache.log4j.Appender;
+import org.apache.log4j.Logger;
+import org.apache.log4j.spi.LoggingEvent;
+import org.junit.Test;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+import org.takes.http.Back;
+
+/**
+ * Test case for {@link BkLogged}.
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 0.11
+ */
+public final class BkLoggedTest {
+    /**
+     * BkLogged can log message.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void logsMessage() throws IOException {
+        final Appender appender = Mockito.mock(Appender.class);
+        Mockito.when(appender.getName()).thenReturn("MOCK");
+        Logger.getRootLogger().addAppender(appender);
+        new BkLogged(
+            new Back() {
+                @Override
+                @SuppressWarnings("PMD.UncommentedEmptyMethod")
+                public void accept(final Socket socket) throws IOException {
+                }
+            }, LogWrap.Level.WARN
+        ).accept(new Socket());
+        Mockito.verify(appender)
+            .doAppend(
+                Matchers.argThat(
+                    new ArgumentMatcher<LoggingEvent>() {
+                        @Override
+                        public boolean matches(final Object argument) {
+                            return ((LoggingEvent) argument)
+                                .getRenderedMessage()
+                                .matches("^.*accept.*$");
```

Comment:
@dmzaytsev Well, I would not use the mock at all.
We can create our own implementation of appender (extend ```org.apache.log4j.AppenderSkeleton```) witch will store log entries for us and provide it to us by request.
Tis appender should be enclosed inside some other class, say, ```TestLog``` with can extend ```org.junit.rules.ExternalResource``` and override ```before()``` and ```after()``` methods, where appended can be initialized and closed.
In the end, usage will be very simple:
1) Declare new field in test class:
```
    @Rule
    public final transient TestLog testlog = new TestLog();
```
2) And used in some test:
        MatcherAssert.assertThat(
            this.testlog.log(),
            Matchers.containsString("Some string we are waiting in logs")
        );
```
And that's it.

But, I understand, you did a lot of work in one PR and I proposed you to create a puzzle for another developer to do these changes.
And again, please, avoid such huge PR in future. They are really hard to manage.